Add regression test

prat-gpt opened the pull request.

ElliotFriedman commented:
please add a test for all of the existing mainnetConfig.toml files so we can regression test all of these templates.

prateek105 committed:
[FPS] multisig task tests (#485)

* remove: name and description from task config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

add example task template config files

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

add fps example templates

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

remove name and description, set task nonce to current multisig nonce

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

fmt and add TODO to switch to manual nonce later

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

templates

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

remove safeNonce from mock network config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* checkpoint

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove isNested flag from toml config file

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fmt

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix is nested safe logic

* fix task01

* add Task02

* refactor _getDataToSign function

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: gas template docs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: template examples to run

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: error handle parsing l2 chain configs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fmt

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove unused variable name

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* error handle reading in and parsing task config file

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add breakdown of calldata difference for signers

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove setting scalars to simplify example task

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add duplicate call structure of task 020 in existing tooling

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove setting gas scalars from gas config template

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* note on commenting out failing check and link to diffchecker

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: task config tests

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add forge build before running template

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update docs to only include gas limits and not gas scalars

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove template ci runs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* Update src/fps/FPS.md

Co-authored-by: blaine <blainemalone01@gmail.com>

* Update src/fps/FPS.md

Co-authored-by: blaine <blainemalone01@gmail.com>

* Update src/fps/FPS.md

* fps -> new task simulation tooling

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove unused template files

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: fps.md -> readme.md

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: move away from fps

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* use op types, remove inlined interfaces

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update readme for running tasks

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update: cleanup, make MultisigTask a generic template

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix test

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* delete: task config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update based on PR feedback, remove taskConfig.toml fields, and safeNonce and isNestedSafe

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove unused mock network configs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove opmainnetConfig.toml

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add op set gas limit config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update command to use new run function

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update command to only have mainnet config toml file

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: networkConfigFilePath -> taskConfigFilePath

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fmt

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* natspec, taskStorageWrites -> _taskStorageWrites

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix test

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add additional docs for address registry

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix: nested multisig issue

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add op gas config command

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add task tests

* Revert "add task tests"

This reverts commit 70ed74f71a41956b276796dc36261782459b13f9.

* add tests

* fix: nonce for nested safes use their own nonce and not parent

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update natspec proposal -> task

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix failing single multisig tests

* cleanup debug logs

* fix address registry failing test

* add test for nested hash to approve

* rename public function

* remove build warnings

* pr feedback

* Update src/fps/task/MultisigTask.sol

Co-authored-by: Elliot <34463580+ElliotFriedman@users.noreply.github.com>

* pr feedback

* fix: virtual keyword to allow template variants

* address pr feedback: multicall import

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* delete testnetConfig.toml

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: natspec comment on taskConfigFilePath var

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* address feedback: add comments

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

---------

Signed-off-by: Elliot <elliotfriedman3@gmail.com>
Co-authored-by: Elliot <elliotfriedman3@gmail.com>
Co-authored-by: Elliot <34463580+ElliotFriedman@users.noreply.github.com>
Co-authored-by: blaine <blainemalone01@gmail.com>

xenoliss committed:
sep/028: executed  (#517)

* execute sep-028

* fix: remove CI task.

---------

Co-authored-by: Blaine Malone <blainemalone01@gmail.com>

blmalone committed:
Better validation of storage layout with bit packing (#513)

* fix: Better validation of storage layout with bit packing

* fix: images dir directly in task.

* fix: attesting in docs to additional values.

prat-gpt committed:
add regression tests for all tasks

ElliotFriedman commented:
closing as we have merged regression tests upstream to main
